components/availability/src/Availability.svelte

```diff
@@ -76,6 +81,52 @@
       });
   };
 
+  let ticks: Date[] = [];
+
+  const minimumTickHeight = 30; // minimum pixels between ticks, vertically
+  const slotSizes = [15, 30, 60, 180, 360]; // we only want to show ticks in intervals of 15 mins, 30 mins, 60 mins, 3 hours, or 6 hours.
+
+  $: ticks = generateTicks(
+    clientHeight,
+    days[0].slots.map((s) => s.start_time),
+  );
+
+  // We don't want to show all 96 15-minute intervals unless the user has a real tall screen.
+  // So let's be smart about it and filter on modification of start_time, end_time, slot_size, or clientHeight
+  // generateTicks() is a recursive func that retries if a minimumTickHeight threshold is not met
+  // Calculation goal of < 5ms -- console.timers kept in place for future development testing.
+  const generateTicks = (
+    height: number,
+    ticks: Date[],
+    intervalCounter: number = 0,
+  ): Date[] => {
+    // console.time('ticks')
+    const tickIters = slotSizes[intervalCounter];
+
+    // ternary here because timeMinute.every(120) doesnt work, but timeHour.every(2) does.
+    let timeInterval =
+      tickIters > 60
+        ? timeHour.every(tickIters / 60)
+        : timeMinute.every(tickIters);
+
+    ticks = scaleTime()
+      .domain(ticks)
+      .ticks(timeInterval as TimeInterval);
+
+    const averageTickHeight = height / ticks.length;
+
+    if (
+      tickIters < slot_size || // dont show 15-min ticks if slot size is hourly
+      (averageTickHeight < minimumTickHeight && // make sure ticks're at least yea-pixels tall
+        intervalCounter < slotSizes.length) // don't try to keep going if youve reached every 6 hours. Subdividing a day into fewer than 4 parts doesn't yield a nice result.
+    ) {
+      return generateTicks(height, ticks, intervalCounter + 1);
+    } else {
+      // console.timeEnd('ticks')
```

Comment:
🔪

Reply:
Per code comment above, leaving these in here to encourage frequent calc time testing. I'm worried that it'll be easy for us to lose sight of this and bloat the function over time. Quiet safeguard.

Reply:
My bad! Makes sense